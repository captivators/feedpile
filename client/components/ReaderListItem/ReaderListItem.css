.list-item-container {
  border-top: 1px solid lightgrey;
  padding-top: 30px;
  padding-bottom: 30px;
  cursor: pointer;
  display: flex;
}

.list-item-container:hover {
  background-color: #f7f7f7;
}

a {
  text-decoration: none;
  color: black;
}

.thumbnail-image {
  width:  150px;
  border-radius: 3px;
  margin-right: 20px;
}

.article-title {
  color: #333;
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
  line-height: 140%;
  padding-bottom: 8px;
  font-weight: 700;
}

.article-publisher {
  font-family: 'Roboto', sans-serif;
  font-size: 12px;
  color: grey;
  padding-bottom: 8px;
}

.article-summary {
  color: #555;
  font-family: 'Roboto', sans-serif;
  font-size: 14px;
  line-height: 150%;
}

@media (max-width: 600px) {
  .list-item-container {
    display: block;
  }
  .thumbnail-container {
    text-align: center;
    padding-bottom: 10px;
  }
}
