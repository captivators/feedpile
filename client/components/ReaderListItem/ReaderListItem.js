import React from 'react';
import moment from 'moment'
import './ReaderListItem.css';
import {connect} from 'react-redux';
import {setCurrentArticle} from '../../actions'

const ReaderListItem = (props) => {
  const {title, imageSrc, author, date, summary, description} = props.article;

  return(
      <div onClick={() => {
        props.history.push(`/read/${props.article._id}`)
        props.renderReadView(props.article)
      }}>
      <div className='list-item-container'>
        <div className='thumbnail-container'>
          <img className='thumbnail-image' src={imageSrc ? imageSrc:'https://s15.postimg.org/6jredm53v/news.png'}/>
        </div>
        <div className='article-text-container'>
          <h2 className='article-title'>{title}</h2>
          <p className='article-author'>{ author ? 'by ' + author + ', ' + moment(date).fromNow():moment(date).fromNow()}</p>
          <p>{}</p>
          <p className='article-summary'>{summary}</p>
        </div>
      </div>
    </div>
  )
};


export const Unwrapped = ReaderListItem;
export default connect(null, {renderReadView: setCurrentArticle})(ReaderListItem);
