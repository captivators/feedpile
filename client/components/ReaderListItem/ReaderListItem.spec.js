import React from 'react';
import ReaderListItem from './ReaderListItem';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('ReaderListItem snapshot test', () => {
  const article = {
    "title":"Trump Administration Hunts for Easter Eggs, and Senior Staff",
    "publisher":"New York Times",
    "image":"http://oi63.tinypic.com/4sbdpx.jpg",
    "url": "http://www.nytimes.com/video/us/politics/100000005018390/trump-administration-hunts-for-easter-eggs-and-senior-staff.html",
    "description":"The White House is as much as two months behind recent standards for presidential transitions, leaving 90 percent of the positions considered critical to leadership unfilled. It did, however, manage to order the eggs for the Easter egg roll."
  };
  const component = shallow(<ReaderListItem article={article}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
