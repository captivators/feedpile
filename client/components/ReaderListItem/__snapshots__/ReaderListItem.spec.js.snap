// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReaderListItem snapshot test 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="list-item-container"
  >
    <div
      className="thumbnail-container"
    >
      <img
        className="thumbnail-image"
        src="https://s15.postimg.org/6jredm53v/news.png"
      />
    </div>
    <div
      className="article-text-container"
    >
      <h2
        className="article-title"
      >
        Trump Administration Hunts for Easter Eggs, and Senior Staff
      </h2>
      <p
        className="article-author"
      >
        a few seconds ago
      </p>
      <p />
      <p
        className="article-summary"
      />
    </div>
  </div>
</div>
`;
